pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularMedicalReleaseV2.java

```diff
@@ -0,0 +1,283 @@
+import org.broadinstitute.ddp.db.dto.FormSectionMembershipDto;
+import org.broadinstitute.ddp.db.dto.StudyDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.activity.definition.FormActivityDef;
+import org.broadinstitute.ddp.model.activity.definition.FormSectionDef;
+import org.broadinstitute.ddp.model.activity.definition.QuestionBlockDef;
+import org.broadinstitute.ddp.model.activity.definition.question.QuestionDef;
+import org.broadinstitute.ddp.model.activity.revision.RevisionMetadata;
+import org.broadinstitute.ddp.model.activity.types.BlockType;
+import org.broadinstitute.ddp.model.user.User;
+import org.broadinstitute.ddp.util.ConfigUtil;
+import org.broadinstitute.ddp.util.GsonUtil;
+import org.jdbi.v3.core.Handle;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+public class SingularMedicalReleaseV2 implements CustomTask {
+    private static final String DATA_FILE = "patches/medical-record-release_v2.conf";
+    private static final String STUDY_GUID = "singular";
+
+    private Config studyCfg;
+    private Config dataCfg;
+    private Instant timestamp;
+    private String versionTag;
+    private Gson gson;
+    private ActivityDao activityDao;
+    private JdbiActivity jdbiActivity;
+    private JdbiActivityVersion jdbiActVersion;
+    private JdbiFormActivityFormSection jdbiFormActivityFormSection;
+
+    @Override
+    public void init(Path cfgPath, Config studyCfg, Config varsCfg) {
+        File file = cfgPath.getParent().resolve(DATA_FILE).toFile();
+        if (!file.exists()) {
+            throw new DDPException("Data file is missing: " + file);
+        }
+        dataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+
+        if (!studyCfg.getString("study.guid").equals(STUDY_GUID)) {
+            throw new DDPException("This task is only for the " + STUDY_GUID + " study!");
+        }
+
+        this.studyCfg = studyCfg;
+        versionTag = dataCfg.getString("versionTag");
+        timestamp = Instant.now(); //parse(dataCfg.getString("timestamp"));
+        gson = GsonUtil.standardGson();
+
+    }
+
+    // This patch will create a new version of medical release for Singular
+    // Approach here is that will read from file a new and complete activity definition from file
+    // We will then create a new version of this activity.
+    // Important: to avoid clashing with existing question stable ids, in new file any question that was mean to
+    // be reused from original activity was named with _NEW prefix.
+    // Once old activity version has been updated so it is deprecated and new activity version has been stored
+    // we will update the new activity version to replace questions with stable id that ends with _NEW with original
+    // question.
+
```

Comment:
This is the story right here. Essentially, there were a lot of changes to the activity (and there might be more coming as we are awaiting IRB approval) so to develop it, I just created a whole activity that I could use instead of medical release v1. The medical_release_v2.conf file is that full activity.
Just found it easier to get it right this way rather than using the Java patch to incrementally update it.
But now I have to read it in and the hack off what we don't want. Read comments in source for more details.